File: 0217-Contains-Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        s = set() # use set to recored appeared num
        for n in nums:
            if n in s:
                return True
            else:
                s.add(n)
        return False
    
    # time complexity O(n)
    # space O(n)

File: 0217_Contains_Duplicate.java
//https://leetcode.com/problems/contains-duplicate/discuss/60858/Possible-solutions.
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> distinct = new HashSet<Integer>(); // add new
        for(int num:nums){
            if (distinct.contains(num)){ // contains in set
                return true;
            }
            distinct.add(num); // add in set
        }
        return false;
    }
}